Add SlabJet analytic data

nilsvu opened the pull request:
## Proposed changes

This PR adds the analytic data for evolving a jet such as the one described in [Komissarov 1999](https://academic.oup.com/mnras/article-abstract/303/2/343/1067242), section 7.4 / Fig. 13.

- [x] Depends on the WENO limiter (cherry-picked from @fmahebert)

A sample run at t=30:

![jet](https://user-images.githubusercontent.com/746230/49591937-16dc7480-f970-11e8-9cb0-2f82ccc6d2a2.png)

### Types of changes:

- [x] New feature

### Component:

- [x] Code

### Code review checklist

- [ ] The PR passes all checks, including unit tests, `clang-tidy` and `IWYU`. For
  instructions on how to perform the CI checks locally refer to the [Dev guide
  on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run `make doc`
  to generate the documentation locally into `BUILD_DIR/docs/html`. Then open
  `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

nilsdeppe commented:
This might be another good test for limiters. Is it actually blocking on anything? I'm sure it won't run super well with minmod, but I don't see that as a reason not to move forward with it :)

nilsvu commented:
@nilsdeppe the current way how boundary conditions are handled require this to have time-dependent functions (despite not being a full solution). Also we can’t have inflow b.c. at the “nozzle” and free b.c. for the rest yet (Note how in the plot above the left boundary is fixed at constant density).

nilsdeppe commented:
can this be rebased and reviewed now?

nilsvu commented:
@nilsdeppe yes, I think so! I'll do it asap

nilsdeppe commented:
Ping on this again. I think we now even have the BCs to support this, but probably not a domain creator+BCs. However, I don't think that should be a blocker to just get the ID merged :)

nilsvu committed:
Add SlabJet analytic data

nilsvu commented:
@nilsdeppe this is now rebased and updated. To reproduce the Komissarov paper exactly we have to add suitable boundary conditions. But here's the result from just evolving the initial data to t=30 with outflow boundary conditions. This means there's a small patch with high-velocity fluid that's moving to the right along magnetic field lines:

t = 0:
<img width="814" alt="Bildschirm­foto 2023-02-11 um 02 30 41" src="https://user-images.githubusercontent.com/746230/218253402-d3403044-8045-4bdb-818c-d3aac0123de5.png">

t = 30:
<img width="799" alt="Bildschirm­foto 2023-02-11 um 02 28 22" src="https://user-images.githubusercontent.com/746230/218253197-fdda7af0-e0e7-4cc6-83e4-0b03af5edc4a.png">

nilsdeppe reviewed (approved):
Thanks for rebasing this! :D
